### DISCUSSION TITLE
How to detect a grid outage

### DISCUSSION THREAD

**Discussion by ViktorJp** [vloschiavo/powerwall2#76] [Q&A] [answered]
First off... thank you VERY much for compiling all this info!  This has been very helpful trying to build a powerwall interface in POSIX shell!  :)

One thing that is still eluding me is to find a way to detect a grid outage.  I'm not seeing anything that would yield those results.  I had high hopes that perhaps there would be something in /api/system_status/grid_faults, but that's just a big mess.

Any ideas?  Thanks much... ;)

**Comment by frood88** [comment, marked as answer]
You should find what you need under:
/api/system_status/grid_status

Where:
“grid_status”:”SystemGridConnected“
Indicates the grid is up and connected

"grid_status":"SystemIslandedActive"
Indicates the grid is down

"grid_status":"SystemTransitionToGrid"
Indicates the grid is up but the Powerwall is not yet in sync with the grid

**Reply by ViktorJp** [reply]
Oh this is perfect... Thank you so much for guiding my way on this!!  :)
